src/modules/authentication/auth-components/ResetPasswordForm.tsx

```diff
@@ -9,53 +10,51 @@ import { useAuth } from 'modules/authentication/hooks';
 
 export const ResetPasswordForm: React.FC = () => {
   const { error } = useSelector((state: AppState) => state.authReducer);
-
-  const dispatch = useDispatch();
-
-  const { resetPassword } = useAuth();
   const [email, setEmail] = useState('');
-  const [loading, setLoading] = useState(true);
-
-  const handleEmail = (e: React.ChangeEvent<HTMLInputElement>) => {
-    const { value } = e.currentTarget;
 
-    setEmail(value);
-    dispatch(AuthAction.error(''));
+  type FormValues = {
+    email: string;
   };
 
-  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
-    e.preventDefault();
+  const dispatch = useDispatch();
 
-    if (email && !error) {
-      resetPassword(email);
-      setLoading(false);
-    }
+  const { register, handleSubmit } = useForm<FormValues>();
 
-    if (error) {
-      setEmail('');
-      setLoading(true);
-    }
+  const { resetPassword } = useAuth();
 
-    console.log(email);
+  const onSubmit = (data: FormValues) => {
+    resetPassword(data.email);
+    setEmail(data.email);
+    console.log(data.email);
     console.log(error);
   };
 
+  const handleEmailChange = () => {
+    dispatch(AuthAction.error(''));
+    setEmail('');
+  };
+
   return (
     <div className='reset-div'>
-      <form onSubmit={handleSubmit} className='reset-form'>
+      <form onSubmit={handleSubmit(onSubmit)} className='reset-form'>
         <label htmlFor='email'>
           Enter the email adress of an account that you want to reset your
           password for:
         </label>
-        <input type='email' value={email} onChange={handleEmail} />
+        <input
+          type='email'
+          {...register('email', { required: true })}
+          onChange={handleEmailChange}
+          id='email-reset'
+        />
         <button type='submit'>Reset password</button>
-        <p className='reset-msg'>
-          {loading
-            ? ''
-            : 'A link for a password reset has been sent to your email adress.'}
-        </p>
-        <p className='reset-err'>{error}</p>
       </form>
+      {!error && email ? (
+        <p>A link for password reset has been sent to your email.</p>
+      ) : (
+        ''
+      )}
+      {error && email ? <p className='reset-err'>{error}</p> : ''}
```

Comment:
Realno loše si složio ovu logiku, mislim da bi najkoncizniji način za ovo napisati bio:

`{email && (error ? <p className='reset-err'>{error}</p> : <p>A link for password reset has been sent to your email.</p>)}`